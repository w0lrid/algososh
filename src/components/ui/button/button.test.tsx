import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from './button';

const text = 'text';

describe('Button component', () => {
  it('renders Button with text', () => {
    const tree = renderer.create(<Button text={text} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders Button without text', () => {
    const tree = renderer.create(<Button />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('disables Button', () => {
    const tree = renderer
      .create(
        <Button
          text={text}
          disabled
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders loading state Button', () => {
    const tree = renderer.create(<Button isLoader />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('handles onClick callback', () => {
    window.alert = jest.fn();

    render(
      <Button
        text={text}
        onClick={() => {
          alert('Callback was called properly');
        }}
      />
    );

    const button = screen.getByText(text);

    fireEvent.click(button);

    expect(window.alert).toHaveBeenCalledWith('Callback was called properly');
  });
});
