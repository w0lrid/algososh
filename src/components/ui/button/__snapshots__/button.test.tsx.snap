// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component disables Button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;

exports[`Button component renders Button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;

exports[`Button component renders Button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button component renders loading state Button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
